scripts/ingests/utils.py

```diff
@@ -721,26 +741,57 @@ def ingest_parallaxes(db, sources, plxs, plx_errs, plx_refs, save_db=False, verb
 
     n_added = 0
 
-    for i, source in enumerate(sources):
+    for i, source in enumerate(sources): # loop through sources with parallax data to ingest
         db_name = db.search_object(source, output_table='Sources')[0]['source']
 
-        # Search for existing parallax data.
+        # Search for existing parallax data and determine if this is the best
         # If no previous measurement exists, set the new one to the Adopted measurement
-        # TODO: figure out adopted in cases of multiple measurements.
         adopted = None
+        duplicate = False
         source_plx_data = db.query(db.Parallaxes).filter(db.Parallaxes.c.source == db_name).table()
+        # TODO: Write a function to deal with setting adopted flag
         if source_plx_data is None or len(source_plx_data) == 0:
+            # if there's no other measurements in the database, set new data Adopted = True
             adopted = True
+            duplicate = False
+            old_adopted = None
+        elif len(source_plx_data) > 0:  # plx data already exists
+            for plx_data in source_plx_data:
+                if plx_data['reference'] == plx_refs[i]:  # check to see if other measurement is a duplicate of the new data
+                    duplicate = True
+                    verboseprint("Duplicate measurement\n", plx_data)
+                if plx_data['adopted'] == 1:  # check if another measurement is adopted
+                    old_adopted = plx_data
+            if not duplicate:
+                verboseprint("!!! Another Proper motion measurement exists,")
+                if verbose:
+                    source_plx_data.pprint_all()
+
+                # if errors of new data are less than other measurements, set Adopted = True.
+                if plx_errs[i] < min(source_plx_data['parallax_error']):
+                    adopted = True
+                    # unset old adopted
+                    if old_adopted:
+                        db.Parallaxes.update().where(and_(db.Parallaxes.c.source == old_adopted['source'],
+                                                             db.Parallaxes.c.reference == old_adopted['reference'])).\
+                            values(adopted=False).execute()
+                        old_adopted_data = db.query(db.Parallaxes).filter(and_(db.Parallaxes.c.source == old_adopted['source'],
+                                                                          db.Parallaxes.c.reference == old_adopted['reference'])).table()
+                        verboseprint("Old adopted measurement unset\n", old_adopted_data)
```

Comment:
Isn't this effectively the same as `old_adopted` above? Or do you have this here so you can check that adopted was reset?

Reply:
Yep, it is a check to see that the adopted flag has changed.